api/model/storage/postgres/timeseries.go

```diff
@@ -171,6 +171,39 @@ func (s *Storage) parseDateTimeTimeseriesForecast(rows pgx.Rows) ([]*api.Timeser
 	return points, nil
 }
 
+// Calculate the Min, Max, and Mean of a list of TimerseriesObservation
+func getMinMaxMean(timeseries []*api.TimeseriesObservation) (api.NullableFloat64, api.NullableFloat64, api.NullableFloat64) {
```

Comment:
As a point of Go style - you would probably just want to return floats here, and then defer the NaN checks and use of the NullableFloat64 to the route handler.  The nullable type is only defined and used to support JSON serialization so you wouldn't want it permeating through other parts of the code.